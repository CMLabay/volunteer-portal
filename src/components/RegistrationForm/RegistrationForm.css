.pass-req{
    font-size:12px;
}
.form-input{
    text-align: left; 
}