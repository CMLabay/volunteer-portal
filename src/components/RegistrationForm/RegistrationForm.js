import React, { Component } from 'react'
import { Button, Input, Required } from '../Utils/Utils'
import AuthApiService from '../../services/auth-api-service'
import './RegistrationForm.css'

export default class RegistrationForm extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {}
  }

  state = { error: null }

  handleSubmit = ev => {
    ev.preventDefault()
    const { full_name, user_name, password } = ev.target
    this.setState({ error: null })
    AuthApiService.postUser({
      email: user_name.value,
      password: password.value,
      full_name: full_name.value,
      type: 'user',
    })
      .then(user => {
        full_name.value = ''
        user_name.value = ''
        password.value = ''
        this.props.onRegistrationSuccess()
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render(){
      const { error } = this.state
      return(
          <form
            className='RegistrationForm'
            onSubmit={this.handleSubmit}
          >
         <div role='alert'>
          {error && <p className='red'>{error}</p>}
        </div>
        <div className='full_name'>
          <label htmlFor='RegistrationForm__full_name'>
            Full name <Required />
          </label>
          <Input
            className="form-input"
            name='full_name'
            type='text'
            required
            id='RegistrationForm__full_name'>
          </Input>
        </div>
        <div className='user_name'>
          <label htmlFor='RegistrationForm__user_name'>
            Email <Required />
          </label>
          <Input
            className="form-input"
            name='user_name'
            type='text'
            required
            id='RegistrationForm__user_name'>
          </Input>
        </div>
        <div className='password'>
          <label htmlFor='RegistrationForm__password'>
            Password <Required />
          </label>
          <Input
            className="form-input"
            name='password'
            type='password'
            required
            id='RegistrationForm__password'>
          </Input>
          <p className="pass-req">Password must be at least 8 characters and include at least one of each: 
          <br/>uppercase letter, lowercase letter, number, and special character. </p>
        </div>
        <Button type='submit'>
          Register
        </Button>
      </form>
      )
    }
  }