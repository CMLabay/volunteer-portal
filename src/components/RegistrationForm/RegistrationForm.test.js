import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import RegistrationForm from './RegistrationForm'
import renderer from 'react-test-renderer';

describe('registration form', () =>{


    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <RegistrationForm>
                </RegistrationForm>
            </BrowserRouter>,
            div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI as expected', () => {
        const tree = renderer
        .create(
            <BrowserRouter>
            <RegistrationForm></RegistrationForm>
            </BrowserRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
    })
})