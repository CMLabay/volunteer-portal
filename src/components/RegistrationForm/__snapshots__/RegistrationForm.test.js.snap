// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registration form renders UI as expected 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="full_name"
  >
    <label
      htmlFor="RegistrationForm__full_name"
    >
      Full name 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input form-input"
      id="RegistrationForm__full_name"
      name="full_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="user_name"
  >
    <label
      htmlFor="RegistrationForm__user_name"
    >
      Email 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input form-input"
      id="RegistrationForm__user_name"
      name="user_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm__password"
    >
      Password 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input form-input"
      id="RegistrationForm__password"
      name="password"
      required={true}
      type="password"
    />
    <p
      className="pass-req"
    >
      Password must be at least 8 characters and include at least one of each: uppercase letter, lowercase letter, number, and special character. 
    </p>
  </div>
  <button
    className="Button "
    type="submit"
  >
    Register
  </button>
</form>
`;
